addons/mail/static/src/discuss/discuss_store_service.js

```diff
@@ -0,0 +1,60 @@
+/* @odoo-module */
+
+import { reactive } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+import { onChange } from "@mail/utils/misc";
+
+export class DiscussStore {
+    /** @type {import("@mail/core/store_service").Store} */
+    storeService;
+    constructor(env, { "mail.store": storeService }) {
+        Object.assign(this, { storeService });
+        this.setup(env);
+    }
+
+    setup(env) {
+        this.env = env;
+        this.lastChannelSubscription = "";
+        this.threads = this.storeService.threads;
```

Comment:
`threads` should stay inside `mail` `store` as they exist in the mail module too

Reply:
As we spoke, I'm gonna define a new model for channel and omit the thread model patch.